### DISCUSSION TITLE
Client proxy settings - detect settings automatically

### DISCUSSION THREAD

**Discussion by MPalatucciEng** [rustdesk/rustdesk#9258] [Feature Request]
Hi, thank you for adding the possibility to set only the proxy in the network settings (https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#hide-security-settings--hide-network-settings--hide-server-settings--hide-proxy-settings). 
As a possible development wouldn't it be more efficient to adopt a solution like the other competitors? I.e. add 3 possible settings:

- no proxy; 
- **Detect settings automatically**;
- use manual proxy (where proxy IP, port, username and password are set if required).

Because in the case of using the client for servicing, the user does not always have the knowledge to set the proxy settings. Detecting the proxy settings automatically would be really great.

**Comment by rustdesk** [comment]
Apologize, I didn't get this. Could you explain with some illustration or something else can help understanding?

**Comment by MPalatucciEng** [comment]
Sure, I mean something like that. This way the client automatically uses the system proxy (if used).

![proxy_settings](https://github.com/user-attachments/assets/82a7ddf7-abe2-4e7b-a9d6-f7640ec72f09)

**Reply by rustdesk** [reply]
Agree with you. What system environments they are detecting?

**Reply by MPalatucciEng** [reply]
They probably use the Windows API to fetch the proxy settings. The client works with Automatically Detect Settings, but they are not shown;
